### PR TITLE
Add conda-forge recipe for unique-uncertainty

### PR DESCRIPTION
**Pull request by mtdhuynh**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by gscebba** [comment]
I confirm I will be a maintainer of the package.

**Comment by mtdhuynh** [comment]
@conda-forge-admin, please ping conda-forge/help-python

File: recipes/unique-uncertainty/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/unique_uncertainty-{{ version }}.tar.gz
+  sha256: b7d22a091e09081e9af6e6e7e55d301eacbd258cf8fdd9109845f33bb55b7792
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/unique-uncertainty/meta.yaml:14]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation 
```

File: recipes/unique-uncertainty/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    - shapely >=2.0,<=2.0.2
+    - statsmodels >=0.13.3,<=0.14.2
+    - tqdm >=4.65,<=4.66.2
+
+test:
+  imports:
+    - unique
+    - unique.pipeline
+  commands:
+    - pip install rdkit==2023.09.1
```
Comment (ocefpaf) [inline recipes/unique-uncertainty/meta.yaml:46]: This is not recommended! If you are doing this just to pass pip-check, please note the issue with the check and comment it out.

**Comment by ocefpaf** [comment]
@conda-forge-admin, please lint

File: recipes/unique-uncertainty/meta.yaml (hunk 2 above)
Comment (mtdhuynh) [reply recipes/unique-uncertainty/meta.yaml]: Hi @ocefpaf,

Thank you for the help! 

I tried this hack-y solution because the `pip check` command was raising the following error (e.g., see [L1104](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1037984&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1104) of the `linux_64` job - though the same error happens in `osx osx_64` and `win_64 win` jobs):

```bash
unique-uncertainty 0.2.0 requires rdkit, which is not installed.
```

However, you can see in the [previous lines that actually the `rdkit` package is correctly installed](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1037984&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1043) in the environment.

Furthermore, the `rdkit` dependency is [explicitly stated in the recipe](https://github.com/mtdhuynh/staged-recipes/blob/711b4d03f6f5339641f85bb468ec111ac0b4590b/recipes/unique-uncertainty/meta.yaml#L32C12-L32C13).

I saw that this issues might be due to conflicting/different names between `pypi` and `conda` distributions (see this [SO answer](https://stackoverflow.com/a/71787513)), but the `rdkit` package has the same naming in both [`pypi`](https://pypi.org/project/rdkit/) and [`conda`](https://anaconda.org/conda-forge/rdkit).

Could you provide further delucidation on this matter? Thanks!

Comment (ocefpaf) [reply recipes/unique-uncertainty/meta.yaml]: This is a corner case, not uncommon, where it is fine to comment the pip-check out. Just add some text above it explaining why. But it is better to not perform pip-check, than risking a faulty test against a PyPI dependency.

Comment (mtdhuynh) [reply recipes/unique-uncertainty/meta.yaml]: Thank you @ocefpaf for the prompt help and advice! I have commented it out and now all checks pass.
